src/Hazelcast.Net/Sql/HazelcastSqlException.cs

```diff
@@ -13,26 +13,117 @@
 using Hazelcast.Exceptions;
+using Hazelcast.Core;
 
 namespace Hazelcast.Sql
 {
+    /// <summary>
+    /// Represents the exception that is thrown by the SQL service in case of an error.
+    /// </summary>
+    [Serializable]
     public class HazelcastSqlException : HazelcastException
     {
-        public Guid ClientId { get; }
+        // NOTE: as per CA1032 we implement all constructors, but... keep them private/internal.
+        #pragma warning disable IDE0051 // Remove unused private members - of course
+
+        // TODO: consider adding values to Data["..."] the way RemoteException does?
 
-        internal int ErrorCode { get; }
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HazelcastSqlException"/> class.
+        /// </summary>
+        [ExcludeFromCodeCoverage]
+        private HazelcastSqlException()
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HazelcastSqlException"/> class.
+        /// </summary>
+        [ExcludeFromCodeCoverage]
+        private HazelcastSqlException(string message)
+            : base(message)
+        { }
 
-        internal HazelcastSqlException(Guid clientId, int errorCode, string message) : base(message)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HazelcastSqlException"/> class.
+        /// </summary>
+        [ExcludeFromCodeCoverage]
+        private HazelcastSqlException(string message, Exception innerException)
+            : base(message, innerException)
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HazelcastSqlException"/> class.
+        /// </summary>
+        private HazelcastSqlException(Guid clientId, int errorCode, string message)
+            : base(message)
         {
             ClientId = clientId;
             ErrorCode = errorCode;
         }
 
-        internal HazelcastSqlException(Guid clientId, SqlErrorCode errorCode, string message) : this(clientId, (int)errorCode, message)
+        // FIXME [Oleksii] what is ClientId?
+        //
+        // the 1st ctor below is used for client-side detected anomalies, i.e. a member
+        // is not involved, and then the id is the client id
+        //
+        // the 2nd ctor below is used when the response to a request contains an error,
+        // in which case the error is deserialized by the codec and it contains the
+        // originating member id, ie the id of the member on which the error occurred
+        //
+        // so ClientId can be two totally different things and that is wrong. besides,
+        // these client/member identifiers could be used outside of SQL
+        // - would it make sense for HazelcastException to always include client id?
+        // - would it make sense for some exceptions to always include member id?
+        //   that would be for RemoteException, really, which is the other remote
+        //   error that the server can raise
```

Comment:
I totally agree with problems around the ambiguity of this. In other clients, we use originatingMemberUuid as the name, but as you write it could either be a client or member UUID. Maybe a generic name like `origin` would be better.

Reply:
Good you caught it - I wanted to update this & forgot, going to update. I *think* I'll go with *two* fields, the client & the member, bearing in mind that one might be emtpy.